src/Compilers/Core/Portable/CommandLine/ErrorLogger.cs

```diff
@@ -88,48 +86,62 @@ internal void LogDiagnostic(Diagnostic diagnostic, CultureInfo culture)
 
         private void WriteLocations(Location location, IReadOnlyList<Location> additionalLocations)
         {
-            _writer.WriteArrayStart("locations");
+            if (location.SourceTree != null)
+            {
+                _writer.WriteArrayStart("locations");
+                _writer.WriteObjectStart(); // location
+                _writer.WriteKey("analysisTarget");
```

Comment:
We have one errorlog/run per csc invocation, i.e. per project/assembly. I don't think it makes sense to write out analysisTarget with the same value on every result in the run.